cmd/zoekt-sourcegraph-indexserver/queue.go

```diff
@@ -391,4 +396,25 @@ var (
 		Name: "index_queue_cap",
 		Help: "The number of repositories tracked by the index queue, including popped items. Should be the same as index_num_assigned.",
 	})
+	metricQueueAge = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "index_queue_age_seconds",
+		Help: "A histogram of the amount of time a popped repository spent sitting in the queue beforehand.",
+		Buckets: []float64{
+			60,     // 1m
+			300,    // 5m
+			1200,   // 20m
+			2400,   // 40m
+			3600,   // 1h
+			10800,  // 3h
+			18000,  // 5h
+			36000,  // 10h
+			43200,  // 12h
+			54000,  // 15h
+			72000,  // 20h
+			86400,  // 24h
+			108000, // 30h
```

Comment:
Adding buckets that track durations past 24 hours seems warranted when we're already seeing repos hover around the 22+hr mark: https://sourcegraph.slack.com/archives/C023ELQLV7F/p1650037236875689?thread_ts=1649941550.692219&cid=C023ELQLV7F